# Occupation.

In [2]:
import pandas as pd
%matplotlib inline
import seaborn as sns
import matplotlib

matplotlib.rcParams['figure.figsize'] = (13, 8) 

In [3]:
df = pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user")
df

user_id|age|gender|occupation|zip_code
0                  1|24|M|technician|85711
1                       2|53|F|other|94043
2                      3|23|M|writer|32067
3                  4|24|M|technician|43537
4                       5|33|F|other|15213
..                                     ...
938                 939|26|F|student|33319
939           940|32|M|administrator|02215
940                 941|20|M|student|97229
941               942|48|F|librarian|78209
942                 943|22|M|student|77841

[943 rows x 1 columns]

Load the data set from here - https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user

**1)**  What's wrong with the above dataframe? After figuring it out, read the data using pd.read_csv() with the right parameters. (Hint: Check the documentation for the ```sep``` parameter.

In [4]:
df = pd.read_csv("https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user", sep="|")
df

user_id  age gender     occupation zip_code
0          1   24      M     technician    85711
1          2   53      F          other    94043
2          3   23      M         writer    32067
3          4   24      M     technician    43537
4          5   33      F          other    15213
..       ...  ...    ...            ...      ...
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

[943 rows x 5 columns]

**2)** How many rows and columns does the dataframe have?


like 943 rows and 5 columns

In [ ]:
len(df.columns)

5

In [ ]:
len(df.index)

943

In [ ]:
df.shape

(943, 5)

**3)** What is the statistical summary for the occupation dataset? What is the total male to female ratio?

In [ ]:
df.describe

<bound method NDFrame.describe of      user_id  age gender     occupation zip_code
0          1   24      M     technician    85711
1          2   53      F          other    94043
2          3   23      M         writer    32067
3          4   24      M     technician    43537
4          5   33      F          other    15213
..       ...  ...    ...            ...      ...
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

[943 rows x 5 columns]>

In [ ]:
# get number of females
# get number of males
df["gender"].value_counts()

M    670
F    273
Name: gender, dtype: int64

In [ ]:
df.gender.value_counts().M

670

In [ ]:
x = df.gender.value_counts().M
y = df.gender.value_counts().F
print(x, ":", y)

670 : 273


In [ ]:
df

user_id  age gender     occupation zip_code
0          1   24      M     technician    85711
1          2   53      F          other    94043
2          3   23      M         writer    32067
3          4   24      M     technician    43537
4          5   33      F          other    15213
..       ...  ...    ...            ...      ...
938      939   26      F        student    33319
939      940   32      M  administrator    02215
940      941   20      M        student    97229
941      942   48      F      librarian    78209
942      943   22      M        student    77841

[943 rows x 5 columns]

 **4)** Sort the mean age of the occupations in ascending order. Which occupation is the youngest in terms of mean age? 

In [ ]:
occupations_dictionary = df.groupby("occupation").age.mean().to_dict()
sort_it = sorted(occupations_dictionary.items(), key=lambda x: x[1], reverse=False)

for i in sort_it:
    print( i[0] , '=' , i[1] )

student = 22.081632653061224
none = 26.555555555555557
entertainment = 29.22222222222222
artist = 31.392857142857142
homemaker = 32.57142857142857
programmer = 33.121212121212125
technician = 33.148148148148145
other = 34.523809523809526
scientist = 35.54838709677419
salesman = 35.666666666666664
writer = 36.31111111111111
engineer = 36.38805970149254
lawyer = 36.75
marketing = 37.61538461538461
executive = 38.71875
administrator = 38.74683544303797
librarian = 40.0
healthcare = 41.5625
educator = 42.01052631578948
doctor = 43.57142857142857
retired = 63.07142857142857


Student

**5)** What is the male to female ratio for each occupation?



In [ ]:
for i in df.groupby("occupation"):
  x = i.where('gender' == 'M')
  y = i.loc[l['gender'] == 'F'].value_counts()
  print(x, ":", y)

AttributeError: ignored

In [ ]:
df["occupation"].sort_index(ascending=False)

942          student
941        librarian
940          student
939    administrator
938          student
           ...      
4              other
3         technician
2             writer
1              other
0         technician
Name: occupation, Length: 943, dtype: object

In [ ]:
x=df.groupby(["occupation","gender"])["gender"].count()
print(x)

occupation     gender
administrator  F          36
               M          43
artist         F          13
               M          15
doctor         M           7
educator       F          26
               M          69
engineer       F           2
               M          65
entertainment  F           2
               M          16
executive      F           3
               M          29
healthcare     F          11
               M           5
homemaker      F           6
               M           1
lawyer         F           2
               M          10
librarian      F          29
               M          22
marketing      F          10
               M          16
none           F           4
               M           5
other          F          36
               M          69
programmer     F           6
               M          60
retired        F           1
               M          13
salesman       F           3
               M           9
scientist      F     

 **6)** What is the maximum and minimum age for each occupation?

In [ ]:
#1 list of minimum ages for occupations in alphabetical order
#2 list of maximum ages for occupations in alphabetical order
#3 combine ages
#4 list of occupations in alphabetical order
#5 nvm u just print it

In [ ]:
suck_it = df.groupby("occupation").age.min()
val = df.groupby("occupation").age.max()
print(suck_it)
print(val)
print("next time use the stuff we learnt")
print("LMAO NERD")

occupation
administrator    21
artist           19
doctor           28
educator         23
engineer         22
entertainment    15
executive        22
healthcare       22
homemaker        20
lawyer           21
librarian        23
marketing        24
none             11
other            13
programmer       20
retired          51
salesman         18
scientist        23
student           7
technician       21
writer           18
Name: age, dtype: int64
occupation
administrator    70
artist           48
doctor           64
educator         63
engineer         70
entertainment    50
executive        69
healthcare       62
homemaker        50
lawyer           53
librarian        69
marketing        55
none             55
other            64
programmer       63
retired          73
salesman         66
scientist        55
student          42
technician       55
writer           60
Name: age, dtype: int64
next time use the stuff we learnt
LMAO NERD
